### ISSUE TITLE
Redirection to S3 fails with HTTP 400

### ISSUE THREAD

**Issue by aselafernando** [ddurdle/Amazon-Cloud-Drive-for-KODI#33]
If you attempt to view a large file (I've tried in the order of 30GB) Amazon gives a HTTP 302 redirect to the plugin. This redirect is direct to the file stored in AWS S3 storage rather than going through the Amazon cloud drive proxy.

Because the authentication is built into the URL accessing S3, providing the Authorization header when accessing the S3 link returns a HTTP 400 error.

This can be fixed by implementing a redirection callback to modify the headers. You could make it smarter by detecting if it was going to S3 but Amazon do not seem to use 3XX redirects when accessing the cloud drive for any other purpose.

RedirectHandler.py
```python
from urllib2 import HTTPRedirectHandler
from urllib2 import Request

class RedirectHandler(HTTPRedirectHandler):
    def redirect_request(self, req, fp, code, msg, headers, newurl):
        if code in (301, 302, 303, 307):
            newheaders = dict((k,v) for k,v in req.headers.items()
                              if k.lower() not in ("authorization"))
            return Request(newurl, headers=newheaders)
        else: 
            raise urllib2.HTTPError(req.get_full_url(), code, msg, headers, fp)
```

Other Files that need to accessing the cloud media files (cloudservice.py)
```python
from RedirectHandler import RedirectHandler
...
opener = urllib2.build_opener(RedirectHandler())
#Global option to handle all url requests:
urllib2.install_opener(opener)
#Or per request basis
opener.open(**urllib2 Request**)
...
```

**Comment by ddurdle** [comment]
Thanks, I'll integrate this change.

**Comment by aselafernando** [comment]
Hey @ddurdle I thought I'd just add the fix into #35 along with the fix for the additional header issue

**Comment by ddurdle** [comment]
promoted to stable
